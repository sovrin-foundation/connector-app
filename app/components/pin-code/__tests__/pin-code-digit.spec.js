// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import PinCodeDigit from '../pin-code-digit'
import type { PinCodeDigitProps } from '../type-pin-code-box'

describe('<PinCodeDigit />', () => {
  let onPress
  beforeEach(() => {
    onPress = jest.fn()
  })

  it('when digit is entered', () => {
    const wrapper = renderer
      .create(<PinCodeDigit entered onPress={onPress} />)
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('when digit is not entered', () => {
    const wrapper = renderer
      .create(<PinCodeDigit entered={false} onPress={onPress} />)
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
