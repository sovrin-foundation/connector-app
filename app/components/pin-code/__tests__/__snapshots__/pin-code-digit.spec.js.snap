// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PinCodeDigit /> when digit is entered 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  center={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#777777",
      "borderBottomWidth": 5,
      "height": 66,
      "justifyContent": "center",
      "marginHorizontal": 5,
      "width": 38,
    }
  }
  testID={undefined}
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 26,
        "width": 26,
      }
    }
  />
</View>
`;

exports[`<PinCodeDigit /> when digit is not entered 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  center={true}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#777777",
      "borderBottomWidth": 5,
      "height": 66,
      "justifyContent": "center",
      "marginHorizontal": 5,
      "width": 38,
    }
  }
  testID={undefined}
/>
`;
